types/react-router-dom/index.d.ts

```diff
@@ -7,7 +7,7 @@
 //                 Daniel Nixon <https://github.com/danielnixon>
 //                 Tony Ward <https://github.com/ynotdraw>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.1
```

Comment:
I'm not so sure about this one. It's technically a breaking change. But I wouldn't know how this applies practially. Since you didn't change any typings I don't see why we need to change it.

Reply:
Hey @eps1lon ,

My thought process was that because to make things up to do date, we should up the version of Typescript to only ones that are supported, so choose min(supportedTypescripts). If that's not the case I'm happy to remove that change. This is to ensure none fills the bug against non supported typescript version.

Reply:
> This is to ensure none fills the bug against non supported typescript version.

But 2.8 is currently supported. I'm not aware of bug reports that only apply to 2.8

Reply:
@eps1lon Can you then explain me what README below means? I thought this is supported versions:
<img width="868" alt="Screenshot 2020-09-12 at 12 27 36 PM" src="https://user-images.githubusercontent.com/759579/92993644-6e1f5980-f4f3-11ea-83aa-9c7a9174850b.png">

It clearly states that support for 2.8 ended in March 2020. If that's not the case where I can find correct information?

Reply:
DT maintainer here: Only 3.1 and above are supported; you don't need this comment anymore if you support 3.1 and above.

(and, uh, looking at that schedule, looks like I should remove 3.1 this week too.)

Reply:
>  2.8 was supported, be aware that DT itself no longer tests that 2.8

@sandersn I wasn't aware of that. Shouldn't the test runner throw an error once the underlying types are changed since the comment is misleading?

Reply:
Yeah, that's not a bad idea. I filed https://github.com/microsoft/dtslint/issues/309 to track this.

Reply:
Though it's technically not tested anyway I'd still argue to keep the documented TS version at 2.8. The React core team asked us specifically to not make any type related breaking changes for v17. While that's always possible with types it's less likely if we keep a human check for the kind of TS we allow (using `unknown` for example).

Especially since changing the documented, minimum supported TS version doesn't unblock anything as far as I can tell.

Reply:
@eps1lon Can you clarify this:
> The React core team asked us specifically to not make any type related breaking changes for v17.

Does the change on [Line 62 below](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47532/files#diff-5e9e7e642af8d89c125bd50d8354b670R62) constitute a type-related breaking change that should be avoided? If so, should this PR not be merged with that change?

Reply:
Oh no, in my mind we were talking about `react` not `react-router-dom`. Discard all that was said. It applies only to `react` not `react-router-dom`